[superlu] build failure

**Host Environment**
- OS: Windows 11 22621.232
- Compiler: Visual Studio 17 2022

**To Reproduce**
Steps to reproduce the behavior:
```./vcpkg install superlu::x64-windows```

And in CMake
```
[cmake] CMake Error at E:/Downloads/Git/vcpkg/installed/x64-windows/share/superlu/superluTargets.cmake:78 (message):
[cmake]   The imported target "superlu::superlu" references the file
[cmake] 
[cmake]      "E:/Downloads/Git/vcpkg/installed/x64-windows/debug/lib/superlu.lib"
[cmake] 
[cmake]   but this file does not exist.  Possible reasons include:
[cmake] 
[cmake]   * The file was deleted, renamed, or moved to another location.
[cmake] 
[cmake]   * An install or uninstall procedure did not complete successfully.
[cmake] 
[cmake]   * The installation package was faulty and contained
[cmake] 
[cmake]      "E:/Downloads/Git/vcpkg/installed/x64-windows/share/superlu/superluTargets.cmake"
[cmake] 
[cmake]   but not all the files it references.
```
Apparently there is no ```superlu.lib```.

**Failure logs**
```
PS E:\Downloads\Git\vcpkg> .\vcpkg.exe install superlu:x64-windows
Computing installation plan...
The following packages will be built and installed:
    superlu[core]:x64-windows -> 2020-01-07#5
Detecting compiler hash for triplet x64-windows...
-- Automatically setting HTTP(S)_PROXY environment variables to "127.0.0.1:7890".
Restored 0 package(s) from C:\Users\mrs504aa\AppData\Local\vcpkg\archives in 671.3 us. Use --debug to see more details.
Installing 1/1 superlu:x64-windows...
Building superlu[core]:x64-windows...
-- Using cached xiaoyeli-superlu-a3d5233770f0caad4bc4578b46d3b26af99e9c19.tar.gz.
-- Cleaning sources at E:/Downloads/Git/vcpkg/buildtrees/superlu/src/6af99e9c19-18e507a628.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source E:/Downloads/Git/vcpkg/downloads/xiaoyeli-superlu-a3d5233770f0caad4bc4578b46d3b26af99e9c19.tar.gz
-- Applying patch fix-libm.patch
-- Applying patch remove-make.inc.patch
-- Using source at E:/Downloads/Git/vcpkg/buildtrees/superlu/src/6af99e9c19-18e507a628.clean
-- Found external ninja('1.10.2').
-- Configuring x64-windows
-- Building x64-windows-dbg
-- Building x64-windows-rel
-- Installing: E:/Downloads/Git/vcpkg/packages/superlu_x64-windows/share/superlu/copyright
-- Fixing pkgconfig file: E:/Downloads/Git/vcpkg/packages/superlu_x64-windows/lib/pkgconfig/superlu.pc
-- Using cached msys-mingw-w64-i686-pkg-config-0.29.2-3-any.pkg.tar.zst.
-- Using cached msys-mingw-w64-i686-libwinpthread-git-9.0.0.6373.5be8fcd83-1-any.pkg.tar.zst.
-- Using msys root at E:/Downloads/Git/vcpkg/downloads/tools/msys2/9a1ec3f33446b195
-- Fixing pkgconfig file: E:/Downloads/Git/vcpkg/packages/superlu_x64-windows/debug/lib/pkgconfig/superlu.pc
-- Performing post-build validation
-- Performing post-build validation done
Stored binary cache: "C:\Users\mrs504aa\AppData\Local\vcpkg\archives\4a\4ad7f362a516446e4796d042d56b3af1a52c68406a0dfdebdfd2c7a2a7681c8a.zip"
Elapsed time to handle superlu:x64-windows: 8.921 s

Total elapsed time: 11.03 s

superlu provides CMake targets:
    # this is heuristically generated, and may not be correct
    find_package(superlu CONFIG REQUIRED)
    target_link_libraries(main PRIVATE superlu::superlu)
```

**Additional context**
I tried to install superlu because the default installation of armadillo does not ship with superlu.

--- a/ports/superlu/portfile.cmake
+++ b/ports/superlu/portfile.cmake
@@ -23,9 +23,8 @@ vcpkg_cmake_configure(
 
 vcpkg_cmake_install()
 vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/${PORT})
+vcpkg_fixup_pkgconfig()
 
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
 
 file(INSTALL "${SOURCE_PATH}/License.txt" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
-
-vcpkg_fixup_pkgconfig()

--- a/ports/superlu/portfile.cmake
+++ b/ports/superlu/portfile.cmake
@@ -1,3 +1,7 @@
+if(VCPKG_TARGET_IS_WINDOWS)
+    vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
+endif()
+
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO xiaoyeli/superlu
@@ -9,21 +13,19 @@ vcpkg_from_github(
       remove-make.inc.patch
 )
 
-vcpkg_configure_cmake(
-    SOURCE_PATH ${SOURCE_PATH}
-    PREFER_NINJA
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
     OPTIONS
     -DXSDK_ENABLE_Fortran=OFF
     -Denable_tests=OFF
     -Denable_blaslib=OFF
 )
 
-vcpkg_install_cmake()
-vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/${PORT})
-vcpkg_copy_pdbs()
+vcpkg_cmake_install()
+vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/${PORT})
 
-file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
 
-file(INSTALL ${SOURCE_PATH}/License.txt DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
+file(INSTALL "${SOURCE_PATH}/License.txt" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
 
 vcpkg_fixup_pkgconfig()

--- a/ports/superlu/vcpkg.json
+++ b/ports/superlu/vcpkg.json
@@ -4,8 +4,8 @@
   "port-version": 6,
   "description": "Supernodal sparse direct solver.",
   "homepage": "https://github.com/xiaoyeli/superlu",
-  "supports": "!(uwp | arm)",
   "license": "BSD-4-Clause-UC",
+  "supports": "!(uwp | arm)",
   "dependencies": [
     "blas",
     {

--- a/ports/superlu/vcpkg.json
+++ b/ports/superlu/vcpkg.json
@@ -1,11 +1,20 @@
 {
   "name": "superlu",
   "version-date": "2020-01-07",
-  "port-version": 5,
+  "port-version": 6,
   "description": "Supernodal sparse direct solver.",
   "homepage": "https://github.com/xiaoyeli/superlu",
   "supports": "!(uwp | arm)",
+  "license": "BSD-4-Clause-UC",
   "dependencies": [
-    "blas"
+    "blas",
+    {
+      "name": "vcpkg-cmake",
+      "host": true
+    },
+    {
+      "name": "vcpkg-cmake-config",
+      "host": true
+    }
   ]
 }

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -6946,7 +6946,7 @@
     },
     "superlu": {
       "baseline": "2020-01-07",
-      "port-version": 5
+      "port-version": 6
     },
     "symengine": {
       "baseline": "0.9.0",

--- a/versions/s-/superlu.json
+++ b/versions/s-/superlu.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "11fd793ec04d8f07b56b75a5b9938c355bf3f144",
+      "git-tree": "6b8a53560265803d6b0b5b284b88abf0ae48650d",
       "version-date": "2020-01-07",
       "port-version": 6
     },

--- a/versions/s-/superlu.json
+++ b/versions/s-/superlu.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "11fd793ec04d8f07b56b75a5b9938c355bf3f144",
+      "version-date": "2020-01-07",
+      "port-version": 6
+    },
+    {
       "git-tree": "5bee1e0197c0e768c6ee8b9acdf815b4d46b5978",
       "version-date": "2020-01-07",
       "port-version": 5

